components/omega/src/ocn/HorzMesh.h

```diff
@@ -0,0 +1,231 @@
+                                ///  are part of each cellsOnVertex (m^2)
+   ArrayHost2DR8
+       KiteAreasOnVertexH; ///< Area of the portions of each dual cell that
+                           ///  are part of each cellsOnVertex (m^2)
+
+   Array1DR8 DvEdge; ///< Length of each edge, computed as the distance between
+                     ///  verticesOnEdge (m)
+   ArrayHost1DR8 DvEdgeH; ///< Length of each edge, computed as the distance
+                          ///  between verticesOnEdge (m)
+
+   Array1DR8 DcEdge; ///< Length of each edge, computed as the distance between
+                     ///  CellsOnEdge (m)
+   ArrayHost1DR8 DcEdgeH; ///< Length of each edge, computed as the distance
+                          ///  between CellsOnEdge (m)
+
+   Array1DR8 AngleEdge; ///< Angle the edge normal makes with local eastward
+                        ///  direction (radians)
+   ArrayHost1DR8 AngleEdgeH; ///< Angle the edge normal makes with local
+                             ///  eastward direction (radians)
+
+   ArrayHost1DR8 MeshDensityH; ///< Value of density function used to generate a
+                               ///  particular mesh at cell centers
+
+   // Weights
+
+   Array2DR8 WeightsOnEdge; ///< Reconstruction weights associated with each of
+                            ///  the edgesOnEdge
+   ArrayHost2DR8 WeightsOnEdgeH; ///< Reconstruction weights associated with
+                                 ///  each of the edgesOnEdge
+
+   // Coriolis
+
+   Array1DR8 FEdge;      ///< Coriolis parameter at edges (radians s^-1)
+   ArrayHost1DR8 FEdgeH; ///< Coriolis parameter at edges (radians s^-1)
+
+   Array1DR8 FCell;      ///< Coriolis parameter at cell centers (radians s^-1)
+   ArrayHost1DR8 FCellH; ///< Coriolis parameter at cell centers (radians s^-1)
+
+   Array1DR8 FVertex;      ///< Coriolis parameter at vertices (radians s^-1)
+   ArrayHost1DR8 FVertexH; ///< Coriolis parameter at vertices (radians s^-1)
+
+   // Depth
+
+   Array1DR8 BottomDepth;      ///< Depth of the bottom of the ocean (m)
+   ArrayHost1DR8 BottomDepthH; ///< Depth of the bottom of the ocean (m)
+
+   // Edge sign
+
+   Array2DR8 EdgeSignOnCell;      ///< Sign of vector connecting cells
+   ArrayHost2DR8 EdgeSignOnCellH; ///< Sign of vector connecting cells
+
+   Array2DR8 EdgeSignOnVertex;      ///< Sign of vector connecting vertices
+   ArrayHost2DR8 EdgeSignOnVertexH; ///< Sign of vector connecting vertices
+
+   // Methods
+
+   /// Initialize Omega local mesh
+   static int init();
+
+   /// Construct a new local mesh for a given decomposition
+   HorzMesh(const std::string &Name, ///< [in] Name for mesh
+            Decomp *Decomp           ///< [in] Decomposition for mesh
+   );
+
+   /// Destructor - deallocates all memory and deletes a HorzMesh
+   ~HorzMesh();
+
+   /// Deallocates arrays
+   void clear();
+
+   /// Remove mesh by name
+   void erase(std::string InName);
```

Comment:
I think that `clear` and `erase` should be static methods as in `Decomp`.

Reply:
thanks, @mwarusz. I just corrected that.